### PR TITLE
Tickets/dm 33006

### PR DESCRIPTION
**Pull request by tribeiro**

### CODE REVIEW

File: python/lsst/ts/MTAOS/mtaos.py [hunk 1]
```diff
@@ -414,12 +415,22 @@ async def do_runWEP(self, data):
         else:
             # TODO (DM-31365): Remove workaround to visitId being of type long
             # in MTAOS runWEP command.
+            snd_stamp_isot = (
```
Comment (teweitsai) [inline python/lsst/ts/MTAOS/mtaos.py:418]: What are `snd` and `isot`?

**Review by teweitsai — APPROVED** [review]
The update looks good to me.

File: python/lsst/ts/MTAOS/model.py [hunk 2]
```diff
@@ -532,12 +555,34 @@ async def process_comcam(self, intra_id, extra_id, config):
...
+        if wep_process.returncode != 0:
+            stdout, stderr = await wep_process.communicate()
+
+            if len(stdout) > 0:
+                self.log.debug(stdout.decode())
+
+            if len(stderr) > 0:
+                self.log.error(stderr.decode())
+
+            raise RuntimeError(f"Error running pipeline task: {stderr.decode()}")
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/MTAOS/model.py:583]: So, adding the stderr to the runtime error message makes sure it is also sent in the command acknowledgement error, which causes it to appear on the user side of things. Then, the user don't need to look at the log messages to see know what happened. It is a bit of duplicated info but I think it helps users have an initial insight into potential problems.

File: python/lsst/ts/MTAOS/model.py [hunk 3]
```diff
@@ -873,6 +918,19 @@ async def set_ofc_data_values(self, **kwargs):
...
+    async def log_stream(self, stream: asyncio.subprocess.PIPE) -> None:
+        """Log messages from input stream asynchronously.
+
+        Parameters
+        ----------
+        stream : `asyncio.subprocess.PIPE`
+            Output stream pipe to process and log.
+        """
+
+        while True:
```
[the remark this answers is not in the document]
Comment (tribeiro) [reply python/lsst/ts/MTAOS/model.py:930]: Well, it is not that I hate infinity loops, they are certainly useful if used with care. 

For instance, if you have a worker loop, one that will do some work and then wait a bit and do it again, you can certainly specify an exit condition, e.g. "do this until I don't want you to do it anymore". And you know that, if you set the "do_work" flag to false, it should stop in a certain amount of time.

In this particular case there is no exit condition. The loop is waiting for output from the stream and will be waiting forever. So the only thing one can do to stop this is to cancel it, which is what it does on the method that it is used.

**Pull request merged**